index.d.ts

```diff
@@ -680,6 +697,81 @@ declare global {
     FramePrototypingMixin,
     ReactionMixin {}
 
+  interface OpaqueNodeMixin  {
+    readonly id: string
+    readonly absoluteTransform: Transform
+    relativeTransform: Transform
+    x: number
+    y: number
+    readonly width: number
+    readonly height: number
+    readonly removed: boolean
+    remove(): void
+    readonly parent: (BaseNode & ChildrenMixin) | null
+    name: string
+    toString(): string
+
+    getPluginData(key: string): string
+    setPluginData(key: string, value: string): void
+
+    // Namespace is a string that must be at least 3 alphanumeric characters, and should
+    // be a name related to your plugin. Other plugins will be able to read this data.
+    getSharedPluginData(namespace: string, key: string): string
+    setSharedPluginData(namespace: string, key: string, value: string): void
+    setRelaunchData(data: { [command: string]: /* description */ string }): void
+  }
```

Comment:
Can we remove everything in here that already exists in `BaseNodeMixin`, which all node types will already be extending?

Reply:
Done